### PR TITLE
parser: Sort keywords

### LINKED ISSUE #48850: The keywords in parser.y are not sorted properly
## Enhancement

https://github.com/pingcap/tidb/pull/48807#discussion_r1403206616

### PR DESCRIPTION
**Pull request by dveeden**
### What problem does this PR solve?

Issue Number: close #48850 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by dveeden** [comment]
Tested with`TestKeywordsSorting` in #48801 

```go
func TestKeywordsSorting(t *testing.T) {
	for i, kw := range parser.Keywords {
		if i > 1 && parser.Keywords[i-1].Word > kw.Word && parser.Keywords[i-1].Reserved == kw.Reserved {
			t.Errorf("%s should come after %s, please update parser.y and re-generate keywords.go\n",
				parser.Keywords[i-1].Word, kw.Word)
		}
	}
}
```

**Review by tangenta — APPROVED** [review]

File: pkg/parser/parser.y [hunk 1]
```diff
@@ -281,8 +282,9 @@ import (
 	trailing          "TRAILING"
 	trigger           "TRIGGER"
 	trueKwd           "TRUE"
-	unique            "UNIQUE"
+	tidbCurrentTSO    "TiDB_CURRENT_TSO"
```
Comment (mjonss) [inline pkg/parser/parser.y:285]: Should it be consistend and be all upper case?
```suggestion
	tidbCurrentTSO    "TIDB_CURRENT_TSO"
```

**Review by mjonss — APPROVED** [review]
LGTM, just one question about consistency with all upper case keywords.

File: pkg/parser/parser.y (hunk 1 above)
Comment (dveeden) [reply pkg/parser/parser.y:285]: Yes, maybe. But that's outside of the scope of my PR as the objective of my PR is to reorder without changing anything else.

**Comment by hawkingrei** [comment]
@dveeden It needs @eason or @yudongusa‘s approve

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**